File: swift/Sources/sdnx/Types/DefSchema.swift
import Foundation

public struct DefSchema: SchemaValue {
    public let type: String = "def"
    public var description: String?
    public var validators: [String: ValidatorInfo]?
    public let name: String
    public let inner: Schema
    
    public init(name: String, inner: Schema) {
        self.name = name
        self.inner = inner
    }
}

File: dotnet/Sdnx.Core/Types/DefSchema.cs
namespace Sdnx.Core
{
    public record DefSchema : SchemaValue
    {
        public required string Type { get; init; } = "def";
        public required string Name { get; init; }
        public required Dictionary<string, SchemaValue> Inner { get; init; }
        public string? Description { get; init; }
        public Dictionary<string, Validator>? Validators { get; init; }
    }
}
